include/kodlab_mjbots_sdk/robot_interface.h

```diff
@@ -0,0 +1,143 @@
+/*!
+ * @file robot_interface.h
+ * @author Kodlab - J. Diego Caporale (jdcap@seas.upenn.edu)
+ * @brief Robot Interface Class 
+ * @date 2022-07-12
+ * 
+ * @copyright Copyright (c) 2021 The Trustees of the University of Pennsylvania. All Rights Reserved
+ *            BSD 3-Clause License
+ * 
+ */
+#pragma once 
+
+#include <vector>
+#include <memory>
+#include "kodlab_mjbots_sdk/common_header.h"
+#include "kodlab_mjbots_sdk/joint_base.h"
+#include "kodlab_mjbots_sdk/soft_start.h"
+
+namespace kodlab
+{
+    /*!
+     * @brief Sets up a kodlab robot interface class 
+     * 
+     */
+    class RobotInterface {
+    public:
+        static const int KILL_ROBOT = -1;
+        std::vector< std::shared_ptr<JointBase> > joints; ///the vector of shared_ptrs to joints 
+        u_int64_t cycle_count_; //TODO Make this time based not cycle based for when the system fails to keep up (i.e. time_us_)
+        
+        /*!
+         * @brief constructs a robot_interface that contains the basic state of a jointed robot with attitude
+         * @param joint_vect a vector of shared pointers to jointbase defining the motors in the robot
+         * @param soft_start_duration how long in dt to spend ramping the torque
+         * @param robot_max_torque the maximum torque to allow per motor in the robot
+         */
+        template <class JointDerived = JointBase>
+        RobotInterface( std::vector<std::shared_ptr<JointDerived>> joint_vect, 
+                        float robot_max_torque,
+                        int soft_start_duration )
+                        : soft_start_(robot_max_torque, soft_start_duration)
+        {
+            // Ensure at compile time that the template is JointBase or a child of JointBase
+            static_assert(std::is_base_of<JointBase, JointDerived>::value);
+
+            // Set up joint vector and joint states
+            for (std::shared_ptr<JointDerived> j : joint_vect){
+                joints.push_back(j); // Copy constructed, uses an implicit upcasting
+                positions_.push_back( j->get_position_reference() );
+                velocities_.push_back( j->get_velocity_reference() );
+                torque_cmd_.push_back( j->get_servo_torque_reference() ); 
+            }
```

Comment:
Maybe here create a map from joint name to joint_id.

Reply:
I think that is a good idea. I'm moving it to an issue (#28)  to be done in it's own PR or with the joint base name  PR (#27).